de.metas.contracts/src/main/java/de/metas/contracts/pricing/trade_margin/CustomerTradeMarginPricingRule.java

```diff
@@ -0,0 +1,175 @@
+				.build();
+
+		if ( !customerTradeMarginService.getCustomerTradeMarginForCriteria(customerTradeMarginSearchCriteria).isPresent() )
+		{
+			logger.debug("Not applying due to missing customer trade margin settings!");
+			return false;
+		}
+
+		return true;
+	}
+
+	@Override public void calculate(IPricingContext pricingCtx, IPricingResult result)
+	{
+		final BPartnerId customerId = pricingCtx.getBPartnerId();
+
+		final BPartnerId salesRepId = Services.get(IBPartnerBL.class).getBPartnerSalesRepId(customerId);
+
+		final CustomerTradeMarginSearchCriteria customerTradeMarginSearchCriteria =
+				CustomerTradeMarginSearchCriteria.builder()
+						.customerId(customerId)
+						.salesRepId(salesRepId)
+						.requestedDate( pricingCtx.getPriceDate() )
+						.build();
+
+		final Optional<CustomerTradeMarginSettings> customerTradeMarginSettings =
+				customerTradeMarginService.getCustomerTradeMarginForCriteria(customerTradeMarginSearchCriteria);
+
+		if ( !customerTradeMarginSettings.isPresent() ) {
+			return;
+		}
+
+		final ProductPrice priceBeforeApplyingRule = ProductPrice
+				.builder()
+				.productId( pricingCtx.getProductId() )
+				.money( Money.of( result.getPriceStd(), result.getCurrencyId() ) )
+				.uomId( result.getPriceUomId() )
+				.build();
+
+		final Optional<CommissionInstance> forecastCommissionInstance =
+				createForecastCommissionInstanceForOneQtyInPriceUOM(pricingCtx, priceBeforeApplyingRule, salesRepId, customerId);
+
+		if ( !forecastCommissionInstance.isPresent() )
+		{
+			return;
+		}
+		final CommissionContract salesRepCommissionContract = forecastCommissionInstance.get().getConfig().getContractFor( Beneficiary.of(salesRepId) );
+
+		final Optional<CommissionPoints> tradedCommissionPointsPerPriceUOM = customerTradeMarginService
+				.getTradedCommissionPointsFor( customerTradeMarginSettings.get(),
+						                       forecastCommissionInstance.get().getShares(),
+						                       salesRepCommissionContract );
+
+		if ( !tradedCommissionPointsPerPriceUOM.isPresent() )
+		{
+			return;
+		}
+
+		final Optional<Money> tradedCommissionPointsValue = SpringContextHolder.instance.getBean(CommissionPointsService.class)
+				.getCommissionPointsValue(
+					tradedCommissionPointsPerPriceUOM.get(),
+					salesRepCommissionContract.getId(),
+					pricingCtx.getPriceDate() );
+
+		if ( tradedCommissionPointsValue.isPresent() )
+		{
+			final Money customerTradeMarginPerPriceUOM = SpringContextHolder.instance.getBean(MoneyService.class)
+					.convertMoneyToCurrency( tradedCommissionPointsValue.get(), result.getCurrencyId() );
+
+			result.setBaseCommissionPointsPerPriceUOM( forecastCommissionInstance.get().getCurrentTriggerData().getForecastedPoints().toBigDecimal() );
+			result.setTradedCommissionPercent( customerTradeMarginSettings.get().getMarginPercent() );
+
+			result.setPriceStd( priceBeforeApplyingRule.toBigDecimal().subtract( customerTradeMarginPerPriceUOM.toBigDecimal() ) );
```

Comment:
when rewriting the price, please do a 
```java
Loggables.withLogger(logger, Level.DEBUG).addLog(...);
```
I think it makes sense to allso do this when returning without having done anything (further up in htis method).
Might spare us a lot of head-scratching later, when a customer asks us why a particular price was not the way she expected.